### PR TITLE
Zero downtime smoke tests

### LINKED ISSUE #16481: Prototype a way to test 0-downtime upgradability
https://github.com/keycloak/keycloak/discussions/14434#discussion-4394846

This issue should cover smoke tests within base testsuite.

### PR DESCRIPTION
**Pull request by vramik**
Closes #16481

This PR contains a suggestion for zero downtime smoke tests. It uses only REST calls to manipulate with objects. More context could be found https://github.com/keycloak/keycloak/discussions/14434#discussioncomment-4969086

### CODE REVIEW

File: testsuite/integration-arquillian/HOW-TO-RUN.md [hunk 1]
```diff
@@ -826,6 +826,19 @@ because this is not UI testing). For debugging purposes you can override the hea
...
+### Tips & Tricks:
+Although it _should_ work in general, you may experience an exception like this:
+```
+java.lang.RuntimeException: java.net.UnknownHostException: keycloak.127.0.0.1.nip.io: nodename nor servname provided, 
+or not known at org.keycloak.testsuite.util.OAuthClient.doWellKnownRequest(OAuthClient.java:1032)
+at org.keycloak.testsuite.url.DefaultHostnameTest.assertBackendForcedToFrontendWithMatchingHostname(
+DefaultHostnameTest.java:226)
+...
+```
+when running these tests on your local machine. This happens when something on your machine or network is blocking DNS queries to [nip.io](https://nip.io)
```
Comment (hmlnarik) [inline testsuite/integration-arquillian/HOW-TO-RUN.md:838]: Can we use `--Djdk.net.hosts.file=${project.build.testOutputDirectory}/hosts_file` instead?

Reference: https://github.com/keycloak/keycloak/blob/main/quarkus/CONTRIBUTING.md#resolving-dns-names-when-running-tests

Comment (hmlnarik) [reply testsuite/integration-arquillian/HOW-TO-RUN.md:838]: I did not realize this was only text moved. Agreed then, no further action is needed.

**Review by hmlnarik — APPROVED** [review]
Excellent work, @vramik , thank you.

Given the area affected, I'd like to ask @miquelsi to review too.

**Review by miquelsi — APPROVED** [review]
Nice piece of work, thanks @vramik!

**Pull request merged**